tool/tctl/common/plugin/awsic.go

```diff
@@ -0,0 +1,208 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"maps"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+
+	pluginspb "github.com/gravitational/teleport/api/gen/proto/go/teleport/plugins/v1"
+	"github.com/gravitational/teleport/api/types"
+	apicommon "github.com/gravitational/teleport/api/types/common"
+	"github.com/gravitational/teleport/lib/utils"
+	toolcommon "github.com/gravitational/teleport/tool/common"
+)
+
+type awsICArgs struct {
+	cmd                  *kingpin.CmdClause
+	defaultOwners        []string
+	scimToken            string
+	scimURL              string
+	region               string
+	arn                  string
+	useSystemCredentials bool
+	userOrigin           string
+	userLabels           toolcommon.Labels
+	groupNameFilters     []string
+	accountNameFilters   []string
+	accountIDFilters     []string
+}
+
+func (a *awsICArgs) validate() error {
```

Comment:
add validation for region, scimURL field?

Reply:
Re SCIM credential validation: I meant to [ping the SCIM endpoint](https://github.com/gravitational/teleport.e/blob/master/lib/web/plugindescriptor_aws_ic.go#L301) so we know the provided SCIM credential is valid. While doing that, we also cover SCIM [endpoint validation](https://github.com/gravitational/teleport.e/blob/master/lib/cloud/aws/aws.go#L80) that is called in the SCIM client constructor.
Aplogies for not being clear.

Reply:
Ah, gotcha. Will investigate.

Reply:
Done

Reply:
And reverted, because all of the SCIM test code I need to use is in `e`, and OSS builds don't have access to it.